cmd/frontend/internal/auth/userpasswd/lockout.go

```diff
@@ -51,8 +70,159 @@ func (s *lockoutStore) IncreaseFailedAttempt(userID int32) {
+	}
+
+	token, err := jwt.ParseWithClaims(urlToken, &unlockAccountClaims{}, func(token *jwt.Token) (interface{}, error) {
+		return base64.StdEncoding.DecodeString(signingKey)
+	}, jwt.WithValidMethods([]string{jwt.SigningMethodHS512.Name}))
+
+	if err != nil {
+		return false, err
+	}
+
+	if claims, ok := token.Claims.(*unlockAccountClaims); ok && token.Valid && userID == claims.UserID && string(storedToken) == urlToken {
+		return true, nil
+	}
+
+	return false, errors.Newf("provided token is invalid or expired")
+}
+
 func (s *lockoutStore) Reset(userID int32) {
 	key := strconv.Itoa(int(userID))
 	s.lockouts.Delete(key)
 	s.failedAttempts.Delete(key)
 }
+
+var emailTemplates = txemail.MustValidate(txtypes.Templates{
+	Subject: `Unlock your Sourcegraph Cloud account`,
+	Text: `
+You are receiving this email because your Sourcegraph account got locked after too many sign in attempts.
+
+Please, visit this link in your browser to unlock the account and try to sign in again: {{.UnlockAccountUrl}}
+
+This link will expire in {{.ExpiryMinutes}} minutes.
+
+To see our Terms of Service, please visit this link: https://about.sourcegraph.com/terms
+To see our Privacy Policy, please visit this link: https://about.sourcegraph.com/privacy
+
+Sourcegraph, 981 Mission St, San Francisco, CA 94103, USA
+`,
+	HTML: `
+<html>
+<head>
+  <meta name="color-scheme" content="light">
+  <meta name="supported-color-schemes" content="light">
+  <style>
+    body { color: #343a4d; background: #fff; padding: 20px; font-size: 16px; font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji; }
+    .logo { height: 34px; margin-bottom: 15px; }
+    a { color: #0b70db; text-decoration: none; background-color: transparent; }
+    a:hover { color: #0c7bf0; text-decoration: underline; }
+    a.btn { display: inline-block; color: #fff; background-color: #0b70db; padding: 8px 16px; border-radius: 3px; font-weight: 600; }
+    a.btn:hover { color: #fff; background-color: #0864c6; text-decoration:none; }
+    .smaller { font-size: 14px; }
+    small { color: #5e6e8c; font-size: 12px; }
+    .mtm { margin-top: 10px; }
+    .mtl { margin-top: 20px; }
+    .mtxl { margin-top: 30px; }
+  </style>
+</head>
+<body style="font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;">
+  <img class="logo" src="https://storage.googleapis.com/sourcegraph-assets/sourcegraph-logo-light-small.png" alt="Sourcegraph logo">
+  <p>
+  	You are receiving this email because your Sourcegraph account got locked after too many sign in attempts..
+  </p>
+  <p class="mtxl">
+    Please, follow this link in your browser to unlock your account and try to sign in again: <a class="btn mtm" href="{{.UnlockAccountUrl}}">Unlock your Account</a>
+  </p>
```

Comment:
I think it makes sense to also include the link as is, in case the HTML rendering breaks:
```suggestion
  <p class="mtxl">
    Click on the button below to unlock your account and try to sign in again <br><a class="btn mtm" href="{{.UnlockAccountUrl}}">Unlock your Account</a>
  </p>
  <p class="smaller">Or visit this link in your browser: <a href="{{.UnlockAccountUrl}}">{{.UnlockAccountUrl}}</a></p>
```

Reply:
ok

Reply:
added